### PR TITLE
treewide: add meta.mainProgram

### PR DESCRIPTION
**Pull request by ambroisie**
Related PR: NixOS#246386

### CODE REVIEW

File: pkgs/development/node-packages/main-programs.nix [hunk 1]
```diff
@@ -47,6 +47,7 @@
   firebase-tools = "firebase";
   fkill-cli = "fkill";
   fleek-cli = "fleek";
+  flood = "flood";
```
Comment (ambroisie) [inline pkgs/development/node-packages/main-programs.nix:50]: Not technically different from the `pname`, but still getting a warning so despite the comment at the top of the file, I added it.

Comment (Artturin) [reply pkgs/development/node-packages/main-programs.nix:50]: The comment is wrong after https://github.com/NixOS/nixpkgs/pull/246386

Comment (roberth) [reply pkgs/development/node-packages/main-programs.nix:50]: At line 14:

```diff
-  # Packages that provide a single executable whose name differs from the package's `name`.
+  # Packages that provide a single or most prominent executable
```

Comment (Artturin) [reply pkgs/development/node-packages/main-programs.nix:50]: https://github.com/NixOS/nixpkgs/pull/247130

Comment (vcunat) [reply pkgs/development/node-packages/main-programs.nix:50]: This broke eval and is blocking channels:
https://hydra.nixos.org/build/230301363/nixlog/1/tail

Comment (vcunat) [reply pkgs/development/node-packages/main-programs.nix:50]: Simple cheap test:
```
nix eval -f. nodePackages.flood.outPath
```

Comment (ambroisie) [reply pkgs/development/node-packages/main-programs.nix:50]: Ah. I had tested `nix build .#nodePackages.flood` and that seems to work.

Comment (ambroisie) [reply pkgs/development/node-packages/main-programs.nix:50]: In fact, `nix eval -f. nodePackages.flood.outPath` works on my tree, so something might have broken when merging? :thinking:

Comment (ambroisie) [reply pkgs/development/node-packages/main-programs.nix:50]: Yup, it's the interaction with #245435 :upside_down_face:.

Comment (ambroisie) [reply pkgs/development/node-packages/main-programs.nix:50]: Fixed in #247224.

**Pull request merged**